src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -98,11 +108,24 @@ private function createNamedSerializerTags(ContainerBuilder $container, string $
         }
     }
 
-    private function bindDefaultContext(ContainerBuilder $container, array $services, array $defaultContext): void
+    private function bindDefaultContext(ContainerBuilder $container, array $services, array $defaultContext, ?string $circularReferenceHandler, ?string $maxDepthHandler): void
     {
         foreach ($services as $id) {
             $definition = $container->getDefinition((string) $id);
-            $definition->setBindings(['array $defaultContext' => new BoundArgument($defaultContext, false)] + $definition->getBindings());
+
+            $context = $defaultContext;
+            if (str_starts_with((string) $id, 'serializer.normalizer.object')) {
```

Comment:
?
```suggestion
            if (str_starts_with((string) $id, 'serializer.normalizer.object.')) {
```

Reply:
It needs to work for the main service (`serializer.normalizer.object`) as well.